responses/maniphest_search.go

```diff
@@ -0,0 +1,165 @@
+package responses
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/uber/gonduit/entities"
+)
+
+// ManiphestSearchResponse contains fields that are in server response to maniphest.search.
+type ManiphestSearchResponse struct {
+	// Data contains search results.
+	Data []*SearchResponseItem `json:"data"`
+	// Curson contains paging data.
+	Cursor entities.Cursor `json:"cursor,omitempty"`
+}
+
+// ManiphestSearchAttachmentColumnBoardsColumn descrbied a column in "columns" attachment.
+type ManiphestSearchAttachmentColumnBoardsColumn struct {
+	// ID is column identifier.
+	ID int `json:"id"`
+	// Phid is column PHID.
+	Phid string `json:"phid"`
+	// Name is column name.
+	Name string `json:"name"`
+}
+
+// ManiphestSearchAttachmentColumnBoardsColumns is a wrapper for a slice of columns.
+type ManiphestSearchAttachmentColumnBoardsColumns struct {
+	// Columns is collection of columns in an attachment.
+	Columns []*ManiphestSearchAttachmentColumnBoardsColumn `json:"columns,omitempty"`
+}
+
+// ManiphestSearchAttachmentColumnBoards is a wrapper type for columns because sometimes columns is a map and sometimes it is an array.
+type ManiphestSearchAttachmentColumnBoards struct {
+	// ColumnMap is dictionary of columns.
+	ColumnMap map[string]*ManiphestSearchAttachmentColumnBoardsColumns
+	// Columns is collection of columns.
+	Columns []*ManiphestSearchAttachmentColumnBoardsColumn
+}
+
+// UnmarshalJSON parses column data from server response.
+func (b *ManiphestSearchAttachmentColumnBoards) UnmarshalJSON(data []byte) error {
+	if b == nil {
+		return errors.New("boards is nil")
+	}
+	jerr := json.Unmarshal(data, &b.ColumnMap)
+	if jerr == nil {
+		return nil
+	}
+	return json.Unmarshal(data, &b.Columns)
+}
+
+// TaskDescription contains task description data.
+type TaskDescription struct {
+	// Raw is raw task description.
+	Raw string `json:"raw"`
+}
+
+// SearchResponseItem contains information about a particular search result.
+type SearchResponseItem struct {
+	// ID is task identifier.
+	ID int `json:"id"`
+	// Type is task type.
+	Type string `json:"type"`
+	// Phid is PHID of the task.
+	Phid string `json:"phid"`
+	// Fields contains task data.
+	Fields struct {
+		// Name is task name.
+		Name string `json:"name"`
+		// Description is detailed task description.
+		Description *TaskDescription `json:"description"`
+		// AuthorPHID is PHID of task submitter.
+		AuthorPHID string `json:"authorPHID"`
+		// OwnerPHID is PHID of the person who currently assigned to task.
+		OwnerPHID string `json:"ownerPHID"`
+		// Status is task status.
+		Status SearchResultStatus `json:"status"`
+		// Priority is task priority.
+		Priority SearchResultPriority `json:"priority"`
+		// Points is point value of the task.
+		Points interface{} `json:"points"`
```

Comment:
Why it is `interface{}`?

Reply:
Ah, I remember this one. Documentation is not super clear on type so I was not sure what to use.

points | points | Point value of the task.

From here: https://secure.phabricator.com/conduit/method/maniphest.search/

What do you think the type should be?

Reply:
It can be numeric value or `null`. In response it is returned as string, e.g. `"0.5"`. Maybe we should cast to `*float`? What do you think?

Reply:
Updated it to be a json.Number. I think it fits perfectly here.